plan/logical_plan_test.go

```diff
@@ -1101,6 +1101,26 @@ func (s *testPlanSuite) TestValidate(c *C) {
 			sql: "select * from t t1 use index(e)",
 			err: ErrKeyDoesNotExist,
 		},
+		{
+			sql: "select a from t having c2",
+			err: ErrUnknownColumn,
+		},
+		{
+			sql: "select a from t group by c2 + 1 having c2",
+			err: ErrUnknownColumn,
+		},
+		{
+			sql: "select a as b, b from t having b",
+			err: ErrAmbiguous,
+		},
+		{
+			sql: "select a + 1 from t having a",
+			err: ErrUnknownColumn,
+		},
+		{
+			sql: "select a from t having sum(avg(a))",
+			err: ErrInvalidGroupFuncUse,
+		},
```

Comment:
The number of deleted cases is 12 but only added 5 cases.

Reply:
some cases are duplicate